File: virtual/AtCoder/viral-03/05.py
# @uni_kakurenbo
# https://github.com/uni-kakurenbo/competitive-programming-workspace
#
# CC0 1.0  http://creativecommons.org/publicdomain/zero/1.0/deed.ja

# #language PyPy3 #

from sys import setrecursionlimit, stderr, argv
# setrecursionlimit(10**5)
def debug(*args, **opts):
    if argv[-1] == "LOCAL_JUDGE": print(*args, **opts, file=stderr)

n = 1 << 20
q = int(input())

A = [-1] * n
L = [-1] * n

def merge(p):
    if L[p % n] >= 0:
        L[p % n] = merge(L[p % n])
        return L[p % n]
    if A[p % n] == -1:
        A[p % n] = x
        return p
    L[p % n] = merge(p + 1)
    return L[p % n]

for _ in range(q):
    t, x = map(int, input().split())
    if t == 1:
        merge(x)
    if t == 2:
        print(A[x%n])

File: virtual/AtCoder/traP_2025-05-30/05.cpp
/*
 * @uni_kakurenbo
 * https://github.com/uni-kakurenbo/competitive-programming-workspace
 *
 * CC0 1.0  http://creativecommons.org/publicdomain/zero/1.0/deed.ja
 */
/* #language C++ 23 GCC */
// #define DEBUGGER_ENABLED

#include "template/standard.hpp"

void solve();

signed main() {
    debug(__cplusplus);
    int $ = 1;
    // std::cin >> $;
    for(int _ = 0; _ < $; ++_) {
        debug_("Case: #" + std::to_string(_));
        solve();
    }
    return 0;
}

#include "template/warnings.hpp"
void solve() {
    i32 n; input >> n;
    vector<i64> a(n); input >> a;

    i64 base = 0;
    std::set<i64> cut = { 0, 360 };
    for(auto x : a) {
        base += x;
        base %= 360;
        cut.insert(base);
    }

    debug(cut);

    i32 ans = 0;

    auto end = std::prev(cut.end());
    for(auto itr=cut.begin(); itr!=end; ++itr) {
        chmax(ans, *(std::next(itr)) - *itr);
    }

    print(ans);
}
